tests/components/cambridge_audio/conftest.py

```diff
@@ -39,7 +39,25 @@ def mock_stream_magic_client() -> Generator[AsyncMock]:
         client = mock_client.return_value
         client.host = "192.168.20.218"
         client.info = Info.from_json(load_fixture("get_info.json", DOMAIN))
+        client.sources = [
+            Source.from_dict(x)
+            for x in load_json_array_fixture("get_sources.json", DOMAIN)
+        ]
+        client.state = State.from_json(load_fixture("get_state.json", DOMAIN))
+        client.play_state = PlayState.from_json(
+            load_fixture("get_play_state.json", DOMAIN)
+        )
+        client.now_playing = NowPlaying.from_json(
+            load_fixture("get_now_playing.json", DOMAIN)
+        )
         client.is_connected = Mock(return_value=True)
+        client.position_last_updated = client.play_state.position
+        client.unregister_state_update_callbacks = AsyncMock(return_value=True)
+
+        async def mock_state_update_callback():
+            await client.register_state_update_callbacks.call_args[0][0](client)
+
+        client.mock_state_update = Mock(side_effect=mock_state_update_callback)
```

Comment:
Rather make this a separate helper function, this way we don't give the impression that the object supports this out of the box

Reply:
Sounds good. Moved to top of `test_media_player.py`